demo_modules/big_image_carousel/big_image_carousel.js

```diff
@@ -0,0 +1,199 @@
+/* Copyright 2015 Google Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+const ModuleInterface = require('lib/module_interface');
+const debug = require('debug');
+const register = require('register');
+const wallGeometry = require('wallGeometry');
+
+class BigImageCarouselServer extends ModuleInterface.Server {
+  constructor(config) {
+    super();
+    debug('BigImageCarousel Server!', config);
+  }
+}
+
+// Create a polygon extending from x,y by w,h.
+// p5 must be a P5.js instance.
+class BigImageCarouselSketch {
+  constructor(p5, surface, extraArgs) {
+    this.p5 = p5;
+    this.surface = surface;
+    this.tiles = null; 
+
+    this.image_list_config = extraArgs;
+  }
+
+  preload() {
+    var p5 = this.p5;
+
+    console.log('preload');
+
+    this.y_index = this.surface.virtualRect.y / this.surface.virtualRect.h;
+
+    console.log('screen with y_index ' + this.y_index);
+
+    const asset = require('client/asset/asset');
+
+    this.tiles = new Array(); 
```

Comment:
Prefer functional code:

this.tiles = Object.keys(this.image_list_config).map(path => {
  ...
})